### PR TITLE
Get sources list for filter from content_provider table

### LINKED ISSUE #5174: Django's image admin view times out
## Description

The `/admin/api/image/` view doesn't load with production data. You can confirm this by trying to access in staging (duplicates production data) if you have access: https://api-staging.openverse.org/admin/api/image/

I wondered if this was caused due to a heavy DB query and got what was being executed locally (see below). I used the same query in Grafana against the staging and production databases, and the result was returned immediately, so a problem at the DB level was discarded. 

<details>
<summary>query</summary>

```sql
SELECT "image"."id", "image"."created_on", "image"."updated_on", "image"."identifier", "image"."foreign_identifier", "image"."title", 
    "image"."foreign_landing_url", "image"."creator", "image"."creator_url", "image"."thumbnail", "image"."provider", "image"."url",
    "image"."filesize", "image"."filetype", "image"."watermarked", "image"."license", "image"."license_version", "image"."source",
    "image"."last_synced_with_source", "image"."removed_from_source", "image"."view_count", "image"."tags", "image"."category",
    "image"."meta_data", "image"."width", "image"."height", COUNT("nsfw_reports"."id") AS "total_report_count",
    (COUNT("nsfw_reports"."id") - COUNT("nsfw_reports"."decision_id")) AS "pending_report_count",
    MIN("nsfw_reports"."created_at") AS "oldest_report_date"
FROM "image"
    INNER JOIN "nsfw_reports" ON ("image"."identifier" = "nsfw_reports"."identifier")
WHERE "nsfw_reports"."id" IS NOT NULL
GROUP BY "image"."id"
    HAVING (COUNT("nsfw_reports"."id") - COUNT("nsfw_reports"."decision_id")) > 0
ORDER BY 27 DESC, 28 DESC, 29 ASC, "image"."id" DESC;
```

</details>

The query returns a count of 458, which the admin might be trying to load all at once, causing a rendering issue. Instead, the audio view (`admin/api/audio/`), with only 38 audio tracks reported, loads fine. We should limit the number of media items shown on these pages and paginate them.

## Reproduction

1. Login into https://api-staging.openverse.org/admin
2. Go to the view https://api-staging.openverse.org/admin/api/image/
3. See error.

## Screenshots

The **image view** when trying to access it.

The **audio view** loading fine.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #5174 by @krysal

## Description

As mentioned in the issue discussion, this PR optimizes the `source` filter by getting the list from the source table. The provider filter is removed as it involves a similar query and we need to confirm this works first.

## Testing Instructions

Run the API, `just api/up`, and observe the filter changes at:
- http://localhost:50280/admin/api/audio/ and
- http://localhost:50280/admin/api/image/

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog PRs) or the media properties generator (`ov just catalog/generate-docs media-props` for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/api/admin/media_report.py [hunk 1]
```diff
@@ -866,10 +890,6 @@ def is_pending(self, obj):
         "is_pending",
         "media_id",  # used because ``media_obj`` does not render a link
     )
-    list_filter = (
```
Comment (obulat) [inline api/api/admin/media_report.py:869]: Why is this filter removed, @krysal ?

Comment (krysal) [reply api/api/admin/media_report.py:869]: We need to reduce potential issues when rendering the page, and the filters declared this way will result in more `DISTINCT` queries, which are prohibitively expensive in the image table. After confirming we can access the image list, we can re-add these filters optimally (like done here with sources), although I don't see them listed in the admin currently.

**Review by obulat — APPROVED** [review]
It's disappointing that the default filter query is so expensive!

This is the query I get for images locally (and I hope if will be faster):
```
SELECT 
    "image"."id",
    "image"."created_on",
    "image"."updated_on",
    "image"."identifier",
    "image"."foreign_identifier",
    "image"."title",
    "image"."foreign_landing_url",
    "image"."creator",
    "image"."creator_url",
    "image"."thumbnail",
    "image"."provider",
    "image"."url",
    "image"."filesize",
    "image"."filetype",
    "image"."watermarked",
    "image"."license",
    "image"."license_version",
    "image"."source",
    "image"."last_synced_with_source",
    "image"."removed_from_source",
    "image"."view_count",
    "image"."tags",
    "image"."category",
    "image"."meta_data",
    "image"."width",
    "image"."height",
    COUNT("nsfw_reports"."id") AS "total_report_count",
    (COUNT("nsfw_reports"."id") - COUNT("nsfw_reports"."decision_id")) AS "pending_report_count",
    MIN("nsfw_reports"."created_at") AS "oldest_report_date"
FROM 
    "image"
INNER JOIN 
    "nsfw_reports" 
ON 
    ("image"."identifier" = "nsfw_reports"."identifier")
WHERE 
    "nsfw_reports"."id" IS NOT NULL
GROUP BY 
    "image"."id"
HAVING 
    (COUNT("nsfw_reports"."id") - COUNT("nsfw_reports"."decision_id")) > 0
ORDER BY 
    "total_report_count" DESC, 
    "pending_report_count" DESC, 
    "oldest_report_date" ASC, 
    "image"."id" DESC;
```

**Comment by krysal** [comment]
@obulat, you can check how fast it is in Grafana. Thanks for the review!

**Pull request merged**